backend/src/tasks/lookingGlass.ts

```diff
@@ -0,0 +1,295 @@
+      });
+      domains.push(currentDomain);
+    }
+  }
+  await saveDomainsToDb(domains);
+
+  const pulledDomains = Domain.createQueryBuilder('domain')
+    .leftJoinAndSelect('domain.organization', 'organization')
+    .andWhere('ip IS NOT NULL')
+    .andWhere('domain.organization=:org', { org: organizationId });
+
+  return pulledDomains.getMany();
+}
+
+export const handler = async (commandOptions: CommandOptions) => {
+  const { organizationId, organizationName, scanId } = commandOptions;
+
+  console.log('Running lookingGlass on organization', organizationName);
+
+  const collections: LGCollectionsResponse[] = await getCollectionForCurrentWorkspace();
+  const vulnerabilities: Vulnerability[] = [];
+  const domainDict: {} = {};
+
+  for (const line of collections) {
+    // Match the organization to the LookingGlass Collection
+    if (organizationName === line.name) {
+      const collectionID = line.id;
+
+      // Query LookingGlass for the threat info
+      const data: LGThreatResponse = await getThreatInfo(collectionID);
+      const responseDomains: Domain[] = await saveAndPullDomains(
+        data,
+        organizationId!,
+        scanId
+      );
+      // Only pull domains that have been seen in the last two months.
+      const twoMonthsAgo = new Date();
+      twoMonthsAgo.setMonth(twoMonthsAgo.getMonth() - 2);
+      console.log(`Pulling domains seen since ${twoMonthsAgo}`);
+      for (const l of data.results) {
+        // Create a dictionary of relevant fields from the API request.
+        const lastSeen = new Date(l.lastSeen);
+        if (lastSeen > twoMonthsAgo) {
+          const val: ParsedLGResponse = {
+            firstSeen: new Date(l.firstSeen),
+            lastSeen: lastSeen,
+            sources: l.sources,
+            ref_type: l.ref.type,
+            ref_right_type: l.ref.right.type,
+            ref_right_id: l.ref.right.id,
+            ref_left_type: l.ref.left.type,
+            ref_left_id: l.ref.left.id,
+
+            right_ticScore: l.right.ticScore,
+            right_classifications: l.right.classifications,
+            right_name: l.right.name,
+
+            left_type: l.left.type,
+            left_ticScore: l.left.ticScore,
+            left_name: l.left.name,
+            vulnOrMal:
+              l.right.classifications[0] === 'Vulnerable Service'
+                ? 'Vulnerability'
+                : 'Malware'
+          };
+
+          // If we havn't seen this domain yet, create a empty dictionary under the domain
+          if (!domainDict[l.left.name]) {
+            domainDict[l.left.name] = {};
+          }
+          // If we have seen this domain and this threat, merge the two by taking the most recent lastDate and oldest firstDate
+          if (domainDict[l.left.name][l.right.name]) {
+            if (lastSeen > domainDict[l.left.name][l.right.name].lastSeen) {
+              domainDict[l.left.name][l.right.name].lastSeen = val.lastSeen;
+            } else {
```

Comment:
I think in order to get the oldest firstSeen, you have to do this? Please correct me if I'm mistaken though.

```suggestion
            } if (firstSeen < domainDict[l.left.name][l.right.name].firstSeen) {
```